src/Components/Admin/ObjectEditor/object.js

```diff
@@ -0,0 +1,318 @@
+                <div>
+                    Modify Texts
+                    <IconButton
+                        className={this.props.classes.addButton}
+                        aria-label="add"
+                        onClick={this.addText}
+                    >
+                        <AddIcon />
+                    </IconButton>
+                </div>
+                <div>
+                    {this.state.texts.map((item) => {
+                        return (
+                            <div key={item.index}>
+                                <h4>
+                                    Text {item.index + 1} of{" "}
+                                    {this.state.texts.length}
+                                </h4>
+                                <p>{item.text}</p>
+                                <IconButton
+                                    onClick={() =>
+                                        this.onMoveUpClick(item.index)
+                                    }
+                                >
+                                    <KeyboardArrowUp />
+                                </IconButton>
+                                <IconButton
+                                    onClick={() =>
+                                        this.onMoveDownClick(item.index)
+                                    }
+                                >
+                                    <KeyboardArrowDown />
+                                </IconButton>
+                                <IconButton
+                                    onClick={() => this.deleteText(item.index)}
+                                    disabled={this.state.texts.length === 1}
+                                >
+                                    <DeleteForever />
+                                </IconButton>
+                            </div>
+                        );
+                    })}
+                </div>
+                <div>
+                    <Button color="primary" onClick={() => this.handleSubmit()}>
+                        Save
+                    </Button>
+                </div>
+            </div>
+        );
+    }
+}
+
+export default function ObjectEditor({
+    match: {
+        params: { sceneId, objectId },
+    },
+}) {
+    const classes = useStyles();
+    const handleError = useErrorHandler();
+    const [puzzleBody, setPuzzleBody] = useState({});
+    const [puzzleType, setPuzzleType] = useState("");
+    const [animationsJson, setAnimationsJson] = useState({});
+    const [isInteractable, setIsInteractable] = useState(false);
+
+    const puzzleTypeList = [
+        { value: "text-pane", label: "Text Puzzle" },
+        { value: "rotation-controls", label: "Rotation Puzzle" },
+        { value: "keypad", label: "Keypad Puzzle" },
+        { value: "visual-pane", label: "Visual Puzzle" },
+        { value: "jigsaw-puzzle", label: "Jigsaw Puzzle" },
+        { value: "ordered-puzzle", label: "Ordered Puzzle" },
+    ];
```

Comment:
Is it possible to convert this into some sort of enum or another constant variable?

Reply:
Tried to use enum, but got an error saying " 'enum' declarations can only be used in TypeScript files. "